policy/centos7/rke2.te

```diff
@@ -19,3 +19,28 @@ rke2_service_domain_template(rke2_service_db)
 container_manage_lib_dirs(rke2_service_db_t)
 container_manage_lib_files(rke2_service_db_t)
 allow rke2_service_db_t container_var_lib_t:file { map };
+
+########################
+# type rke_logreader_t #
+########################
+gen_require(`
+        type container_runtime_t, unconfined_service_t;
+        type container_log_t;
+        class dir { read search };
+        class file { open read };
+        class lnk_file { getattr read };
+')
+container_domain_template(rke_logreader)
+virt_sandbox_domain(rke_logreader_t)
+corenet_unconfined(rke_logreader_t)
+allow rke_logreader_t container_log_t:dir { open read search };
+allow rke_logreader_t container_log_t:lnk_file { getattr read };
+allow rke_logreader_t container_log_t:file { getattr open read };
+
+################################
+# type rke_container_var_lib_t #
+################################
+type rke_container_var_lib_t;
+files_type(rke_container_var_lib_t)
+allow rke_logreader_t rke_container_var_lib_t:dir search;
+allow rke_logreader_t rke_container_var_lib_t:file { getattr open read };
```

Comment:
I am worried that we break expectations of the docker installation by using this custom type for `/var/lib/docker/containers`. Also, I see no transition(s) in this PR (via macro, or otherwise) to `rke_container_var_lib_t` which would effectively override container-selinux policy (as the file-context bits do explicitly). I don't think policy overrides work well on mutable systems (i.e. if you update container-selinux after installing rke2-selinux we would probably need to make sure rke2-selinux triggers post-install scripts to pick up the restorecon, if such is possible).

Given that `rke_logreader_t` is more privileged than `container_t` would it make sense to just give `rke_logreader_t` the read-only perms to `container_var_lib_t`?

Reply:
My thinking was that we might not want to be overly permissive by granting containers the right to read everything with the `container_var_lib_t` label. Thinking again, it's not really much broader. container-selinux creates an fcontext that applies to `/var/lib/docker(/.*)?` rather than `/var/lib/docker/containers(/.*)?` so I don't *think* they would stomp on each other but simplicity would definitely be better here. So I think you're right, I'll forgo the new file type and test this again.